src/mach/mod.rs

```diff
@@ -24,21 +24,43 @@ pub fn peek(bytes: &[u8], offset: usize) -> error::Result<u32> {
     Ok(bytes.pread_with::<u32>(offset, scroll::BE)?)
 }
 
-#[derive(Debug)]
+#[derive(Copy,Clone,Eq,PartialEq,Hash)]
+pub enum Entrypoint {
+    None,
+    /// The entrypoint was from an `LC_MAIN` load command (added in 10.8) and specifies an offset in
+    /// the `__TEXT` segment.
+    Offset(u64),
```

Comment:
> So I'm wondering if we should relocate the offset based on the vmaddress in the text.

Sure, that makes sense. If `pub entry: u64` is a virtual memory address everywhere else, it can be here too.

> I think the formula is:
>
> `(text_vmaddr - text_offset) + entry_offset`

Yep. Things get more complicated if you're in a context where the image has actually been loaded -- you have to calculate and care about the `slide` of the image -- but this formula will at least be consistent with all the other `vmaddr`s that `goblin` exposes.

At that point we can drop the `enum` and go to a boolean flag :-)